static/js/metrics-explorer.js

```diff
@@ -52,17 +52,23 @@ $(document).ready(function() {
 });
 
 
-function metricsExplorerDatePickerHandler(evt) {
+async function metricsExplorerDatePickerHandler(evt) {
     evt.preventDefault();
     $.each($(".range-item.active"), function () {
         $(this).removeClass('active');
     });
+    var selectedId = $(evt.currentTarget).attr("id");
     $(evt.currentTarget).addClass('active');
-    datePickerHandler($(this).attr('id'), "now", $(this).attr('id'))
+    datePickerHandler(selectedId, "now", selectedId);
+    const queryElement = $('#metrics-queries')
+    const metricNames = await getMetricNames();
+    metricNames.metricNames.sort();
+    queryElement.find('.metrics').val(metricNames.metricNames[0]); 
```

Comment:
We don't need to update the metricsName to the firstvalue. It should remain as it was. We should only update the options for metricsName in the dropdown and also update the tags for the value which is present as the metrics name.

Reply:
@sonamgupta21 if the `metricNames` array is coming empty from backend on a specific time range, then should we remove the existing value of the metrics input?

![image](https://github.com/siglens/siglens/assets/108896805/166bfd82-4f10-4032-a7a7-f42132d5616f)

![image](https://github.com/siglens/siglens/assets/108896805/a9edab93-2f27-4eb6-897b-601ff9cb5815)

Reply:
No, we don't need to remove the input value. We will just update the graph and the dropdown options.